Source/Internal/IGListAdapter+UICollectionView.m

```diff
@@ -57,6 +57,69 @@ - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
 
     return view;
 }
+    
+- (BOOL)collectionView:(UICollectionView *)collectionView canMoveItemAtIndexPath:(NSIndexPath *)indexPath {
+    id <IGListAdapterDataSource> dataSource = self.dataSource;
+    IGAssert(dataSource != nil, @"Found a nil dataSource when requesting canMoveItemAtIndexPath for interactive reordering");
+    
+    NSInteger sectionIndex = indexPath.section;
+    NSInteger itemIndex = indexPath.item;
+    
+    // for ease of implementation, first we check with the listAdapter
+    if ([dataSource respondsToSelector:@selector(listAdapter:canMoveObjectInSection:atIndex:)]) {
+        return [dataSource listAdapter:self canMoveObjectInSection:sectionIndex atIndex:itemIndex];
+    }
+    
+    // then we fall back to the sectionController method
+    IGListSectionController *sectionController = [self sectionControllerForSection:sectionIndex];
+    return [sectionController canMoveItemAtIndex:itemIndex];
+}
+    
+- (void)collectionView:(UICollectionView *)collectionView moveItemAtIndexPath:(NSIndexPath *)sourceIndexPath toIndexPath:(NSIndexPath *)destinationIndexPath {
+    NSInteger sourceItemIndex = sourceIndexPath.item;
+    NSInteger destinationItemIndex = destinationIndexPath.item;
+    NSInteger sourceSectionIndex = sourceIndexPath.section;
+    NSInteger destinationSectionIndex = destinationIndexPath.section;
+    
+    IGListSectionController *sourceSectionController = [self sectionControllerForSection:sourceSectionIndex];
+    IGListSectionController *destinationSectionController = [self sectionControllerForSection:destinationSectionIndex];
+    
+    // this is a move within a section
+    if (sourceSectionController == destinationSectionController) {
+        [self moveInSectionControllerInteractive:sourceSectionController
+                                       fromIndex:sourceItemIndex
+                                         toIndex:destinationItemIndex];
+        return;
+    }
+    
+    // this is a reordering of sections themselves
+    if ([sourceSectionController numberOfItems] == 1
+        && [destinationSectionController numberOfItems] == 1) {
```

Comment:
Great, I think this is the right call. Just need to make sure that it is documented heavily that movement between sections is only allowed when `numberOfItems` == 1.

I don't have the answer, but I wonder if we should be asking the section controller or the collection view for # items (e.g. `[collectionView numberOfItemsInSection:section]`). Like what if there is an item visible, but the data has changed....

Will have to think about it more.